### PR TITLE
bpo-22490: Remove __PYVENV_LAUNCHER__ from environment during launch

### PR DESCRIPTION
**Pull request by ronaldoussoren**
Alternative patch for [bpo-22490](https://www.bugs.python.org/issue22490) that removes the shell environment variable __PYVENV_LAUNCHER__ during interpreter launch because it is only used to communicate between the stub executable in Mac/Tools/pythonw.c and the real interpreter.

This avoids problems when launching the "real" interpreter without going through the stub executable. 

https://bugs.python.org/issue22490

### CODE REVIEW

**Review by jaraco — APPROVED** [review]
Looks good to me. I think it would be nice if, since #9498 didn't fail any tests, there were a documented way (even if just in the ticket), one could verify that the behavior is working as intended (a test that would fail with #9498).

**Comment by gvanrossum** [comment]
I would love to know how to repro the original bug so I can demonstrate this actually fixes it.

Here's what I think should be the repro:
```
/Library/Frameworks/Python.framework/Versions/3.8/bin/python3  -c 'import subprocess; p = subprocess.Popen([".mypy/venv/bin/python3","-m","mypy.dmypy","-h"]).communicate()'
```
The setup is that `.mypy/venv` is a virtualenv that has `mypy.dmypy` installed. This is what failed for the user who reported this.

Unfortunately this does not fail for me (with a framework build from master installed). Would I have to do the install via brew? (I would have to learn about locally testing modified brew recipes, I suppose.)

**Review by gvanrossum — APPROVED** [review]
The patch LG from a code review POV.

**Comment by ned-deily** [comment]
@ronaldoussoren ping?

**Comment by mattip** [comment]
We think this is borking creation of a PyPy virtualenv using cpython host on macOS and is was very difficult to detect, so it would be nice to clean this up. For a test, you could try to install pypy 7.3.0 using cpython, see https://foss.heptapod.net/pypy/pypy/issues/3175. Although now that we know we will probably fix it for PyPy's next release.

File: Mac/Tools/pythonw.c [hunk 1]
```diff
@@ -196,6 +196,15 @@ main(int argc, char **argv) {
             }
         }
 
+	/*
+	 * The environment variable is used to pass the value of real_path
+	 * to the actual python interpreter, and is read by code in
+	 * Python/coreconfig.c.
+	 *
+	 * This way the real interpreter knowns how the user invoked the
```
Comment (nsoranzo) [inline Mac/Tools/pythonw.c]:
```suggestion
	 * This way the real interpreter knows how the user invoked the
```

**Comment by jaraco** [comment]
In cc878cd086, I've resolved the merge conflicts with master. These merge conflicts will almost certainly re-emerge in the backport to 3.7 and maybe 3.8.

Rather than let this languish further, I'm going to merge it.

**Comment by jaraco** [comment]
The test failure is also happening in master, so unrelated to this change.

**Pull request merged**